### ISSUE TITLE
padded-blocks rule - add a multi-line option to this rule

### ISSUE THREAD

**Issue by sscaff1** [eslint/eslint#7775]
**What rule do you want to change?**
padded-blocks

**Does this change cause the rule to produce more or fewer warnings?**
Potentially fewer warning depending on the situation.

**How will the change be implemented? (New option, new default behavior, etc.)?**
Added an option to padded-blocks to specify a minimum number of lines required before the rule takes affect.

**Please provide some example code that this change will affect:**

Currently in JSCS, we have the rule of requirePaddingNewlinesInBlocks. The equivalent rule in seems to be padded-blocks in ESLINT. However, there is no way to specify a minimum number of lines required before the rule takes affect. So here is the problem:

Using JSCS, I can specify `"requirePaddingNewlinesInBlocks": 1" and the following are valid for this rule:
```javascript
function foo() { // valid since more than 1 line between the blocks is required to throw an error
  return bar();
}
function foo() { // valid

  dobar();
  return bar();

}
```

Using ESLINT with the rule `"padded-blocks": [2, "always"]`

```js
function foo() { // not valid according to ESLINT
  return bar();
}
function foo() { // valid

  dobar();
  return bar();

}
```

**What does the rule currently do for this code?**
The rule currently causes a linting error whenever padding is not around the blocks. So I would like to specify a rule like this:

`"padded-blocks": [2, "always"]`

So that we have an equivalent for the JSCS rule. Note: PolyJuice currently translate the rule as `"padded-blocks": [2, "always"]`. 

**What will the rule do after it's changed?**

In the example above, the linting error for the ESLINT example will be removed.

**Comment by platinumazure** [comment]
(Noting that this is related to, but hopefully not a duplicate of, #7356. I had [recommended in chat](https://gitter.im/eslint/eslint?at=5852baa4af6b364a29b7ca77) that this could probably be done as an atomic proposal rather than waiting for #7356 to get moving. If anyone on the core team disagrees, I apologize for causing any confusion.)

Hi @sscaff1, thanks for the issue.

I think this behavior should definitely be behind a new option. Maybe something like this? `"padded-blocks": [2, { "withMoreLinesThan": 1 }]` (But please let's find a better option name)

**Comment by platinumazure** [comment]
Oh, and I'll champion this.

**Comment by not-an-aardvark** [comment]
How about `minLines`?

**Comment by cookiengineer** [comment]
Yes, I would prefer minLines as an option, too.

Our codestyle allows less than two statements (blocks) to require no newline in between. If the statements amount is higher than 2, then a newline is required before and after.

```javascript
// this should be possible with `minLines: 3` from @not-an-aardvark

let whatever = doSomething();
if (whatever === true) {
    whatever.another();
    something_else();
}

let another = doSomething();
if (another === false) {

    something_else();
    dowhatever();
    anotherstatement();

}
```

**Comment by platinumazure** [comment]
Okay, I think we agree that `minLines` is a sensible option name.

Current status: I've volunteered to champion this rule. We need 3 more team members to endorse (:+1:) the original proposal and then we can accept the issue. Then someone can write a PR. (Someone can choose to write a PR earlier, there's just the [hopefully small] risk we won't accept the issue and then the PR would be a waste of time, although the code could certainly be repurposed into a plugin.)

**Comment by jmullo** [comment]
Very much needed.

**Comment by ashclarke** [comment]
If possible, I'd like the capability to require 1 new line padding if a block is multi-line:

```
if (this.thereIsThisOption &&
    this.fairlyLongConditionToo()) {
↵
    this.doSomethingElse();

    this.somethingHereToo = false;
}
```

I was thinking I could achieve it with [`padding-line-between-statements`](http://eslint.org/docs/rules/padding-line-between-statements), but no luck as I think the `multiline-block-like` statement type set for `prev` applies to the statement after the closing brace.

**Comment by platinumazure** [comment]
@eslint/eslint-team Looks like this needs one more :+1: to be accepted! Otherwise, we should probably close this soon.

**Comment by Aladdin-ADD** [comment]
just make sure: would it be conflict with [this proposal](https://github.com/eslint/eslint/issues/7145#issuecomment-281882639)? it seems have been accpeted.

**Comment by platinumazure** [comment]
Great catch @Aladdin-ADD, I think you're right. This might conflict with that other proposal.

@sscaff1 Does [the proposal here](https://github.com/eslint/eslint/issues/7145#issuecomment-281882639) potentially meet your needs?

**Comment by sscaff1** [comment]
Yes it does - thank you

**Comment by platinumazure** [comment]
Thanks! I'll go ahead and close this issue then.
